api/src/main/java/io/strimzi/api/kafka/model/KafkaConnectAuthenticationTls.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+
+/**
+ * Configures the Kafka Connect authentication
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = true,
+        builderPackage = "io.strimzi.api.kafka.model"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class KafkaConnectAuthenticationTls extends KafkaConnectAuthentication {
+    private static final long serialVersionUID = 1L;
+
+    public static final String TYPE_TLS = "tls";
+
+    private CertAndKeySecretSource keys;
```

Comment:
Technically, it is just one. So `key` might be better. Also ... I'm always confused about the difference between key and certificate. Wouldn't `certificate` be better than `key`? Isn't this the whole certificate? Just for consideration.

Reply:
Well technically but you already know about that, key and certificate are two different things or at least I always call just "key", the private key while the "certificate" is the related public key but signed by some CA (or self-signed). This `CertAndKeySecretSource` contains both so, maybe the field should be `certAndKey` :-) but I agree on just `certificate`. We have something similar in the TO with the `cert` field which is a `CertAndKey` type. So, let's go for `certificate` here.

Reply:
@scholzj the problem with naming it `certificate` is the following : 

```
authentication:
      type: tls
      certificate:
        secretName: my-secret
        key: user.key
        certificate: user.crt
```

we have two `certificate' fields in that way :-(

Reply:
Well we also have `trustedCertificates` which is longer :-)